ui/conftest.py

```diff
@@ -1,194 +1,247 @@
-
-
-def pytest_exception_interact(node, call, report):
-    """Attach allure screenshot"""
-    context = False
-    if hasattr(node, "funcargs") and type(node.funcargs) == dict and node.funcargs.get("context"):
-        context = node.funcargs.get("context")
-
-    if report.failed and context and context.pages:
-        for page in context.pages:
-            if page.is_closed():
-                continue
-            try:
-                allure.attach(
-                    page.screenshot(full_page=True),
-                    name="screenshot",
-                    attachment_type=allure.attachment_type.PNG,
-                    extension="png",
-                )
-            except Exception as e:
-                print("Fail to take screenshot: {}".format(e))
-
-# def save_screenshot_on_fail(request: pytest.FixtureRequest, page: Page):
-#     if request.session.testsfailed and not page.is_closed():
-#         allure.attach(
-#             page.screenshot(full_page=True),
-#             name="screenshot",
-#             attachment_type=allure.attachment_type.PNG,
-#             extension="png",
+#             / ext
 #         )
-
-
-def pytest_generate_tests(metafunc: tp.Any) -> None:
-    if "browser_name" in metafunc.fixturenames:
-        browsers = metafunc.config.option.browser or ["chrome"]
-        for browser in browsers:
-            if browser not in ["chrome", "chromium", "firefox", "webkit"]:
-                raise ValueError(
-                    f"'{browser}' is not allowed. Only chromium, firefox, or webkit are valid browser names."
-                )
-        metafunc.parametrize("browser_name", browsers, scope="session")
-
-
-def pytest_configure(config: Config) -> None:
-    config.addinivalue_line(
-        "markers", "skip_browser(name): mark test to be skipped a specific browser"
-    )
-    config.addinivalue_line(
-        "markers", "only_browser(name): mark test to run only on a specific browser"
-    )
-
-
-def _get_skiplist(item: tp.Any, values: tp.List[str], value_name: str) -> tp.List[str]:
-    skipped_values: tp.List[str] = []
-    # Allowlist
-    only_marker = item.get_closest_marker(f"only_{value_name}")
-    if only_marker:
-        skipped_values = values
-        skipped_values.remove(only_marker.args[0])
-
-    # Denylist
-    skip_marker = item.get_closest_marker(f"skip_{value_name}")
-    if skip_marker:
-        skipped_values.append(skip_marker.args[0])
-
-    return skipped_values
-
-
-def pytest_runtest_setup(item: tp.Any) -> None:
-    if not hasattr(item, "callspec"):
-        return
-    browser_name = item.callspec.params.get("browser_name")
-    if not browser_name:
-        return
-
-    skip_browsers_names = _get_skiplist(
-        item, ["chrome", "chromium", "firefox", "webkit"], "browser"
-    )
-
-    if browser_name in skip_browsers_names:
-        pytest.skip("skipped for this browser: {}".format(browser_name))
+#         if not result_path:
```

Comment:
pls remember to remove all not needed commented out code

Reply:
I only commented out all the lines in that file, but I didn't create the file itself